### PR TITLE
minor changes for 4.10.1

### PR DESCRIPTION
**Pull request by MarkWieczorek**
* Update handling of the project version number "4.10" in the web documentations, which was previously treated as an integer.
* remove "-static" option from compiler options.
* Update version numbers of packages used in workflow.
* convert some strings to raw format when they contain latex backslashes.

### CODE REVIEW

File: pyshtools/shclasses/shmaggrid.py [hunk 1]
```diff
@@ -288,7 +288,7 @@ def plot_theta(self, projection=None, tick_interval=[30, 30],
         cb_triangles : str, optional, default = 'neither'
             Add triangles to the edges of the colorbar for minimum and maximum
             values. Can be 'neither', 'both', 'min', or 'max'.
-        cb_label : str, optional, default = '$B_\\theta$, nT'
```
Comment (QuLogic) [inline pyshtools/shclasses/shmaggrid.py:291]: This sort of change won't have any effect as it's just text inside a docstring. You need the `r` on the whole thing, or doubled backslashes.

Comment (MarkWieczorek) [reply pyshtools/shclasses/shmaggrid.py:291]: Thanks! I think I caught the remaining errors.

**Comment by markdjwilliams** [comment]
Is this ready to be merged?

**Comment by MarkWieczorek** [comment]
> Is this ready to be merged?

Hi there, I'll try do this next week (Sept 5+). Sorry for the delay!

**Pull request merged**